<!DOCTYPE html>
<html lang="en">
  <head>
    
      <title>
        T-Cell Receptor Sequencing using Neural Networks ::
        Vansh
      </title>
    
    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<meta
  name="description"
  content="The Greatest Threat to Humanity… What is the greatest threat to humanity? Some might say that it’s a nuclear fallout. Others might say it’s the threat of a large asteroid hitting our planet. In fact, most people believe that the biggest threat to humanity is some out of this world, unimaginable thing that will wipe out humans in one sweep. Well… the greatest threat to humanity is right under our noses."
/>
<meta
  name="keywords"
  content=""
/>
<meta name="robots" content="noodp" />
<link rel="canonical" href="https://vanshs.com/blog/technical/tcellsequencingneuralnetworks/" />





<link rel="stylesheet" href="https://vanshs.com/blog/assets/style.css" />

<link rel="stylesheet" href="https://vanshs.com/blog/style.css" />


<link
  rel="apple-touch-icon-precomposed"
  sizes="144x144"
  href="https://vanshs.com/blog/img/apple-touch-icon-144-precomposed.png"
/>
<link rel="shortcut icon" href="https://vanshs.com/blog/img/favicon.png" />


<link href="https://vanshs.com/blog/assets/fonts/Inter-Italic.woff2" rel="preload" type="font/woff2" as="font" crossorigin="">
<link href="https://vanshs.com/blog/assets/fonts/Inter-Regular.woff2" rel="preload" type="font/woff2" as="font" crossorigin="">
<link href="https://vanshs.com/blog/assets/fonts/Inter-Medium.woff2" rel="preload" type="font/woff2" as="font" crossorigin="">
<link href="https://vanshs.com/blog/assets/fonts/Inter-MediumItalic.woff2" rel="preload" type="font/woff2" as="font" crossorigin="">
<link href="https://vanshs.com/blog/assets/fonts/Inter-Bold.woff2" rel="preload" type="font/woff2" as="font" crossorigin="">
<link href="https://vanshs.com/blog/assets/fonts/Inter-BoldItalic.woff2" rel="preload" type="font/woff2" as="font" crossorigin="">


<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="T-Cell Receptor Sequencing using Neural Networks"/>
<meta name="twitter:description" content="The Greatest Threat to Humanity… What is the greatest threat to humanity? Some might say that it’s a nuclear fallout. Others might say it’s the threat of a large asteroid hitting our planet. In fact, most people believe that the biggest threat to humanity is some out of this world, unimaginable thing that will wipe out humans in one sweep. Well… the greatest threat to humanity is right under our noses."/>



<meta property="og:title" content="T-Cell Receptor Sequencing using Neural Networks" />
<meta property="og:description" content="The Greatest Threat to Humanity… What is the greatest threat to humanity? Some might say that it’s a nuclear fallout. Others might say it’s the threat of a large asteroid hitting our planet. In fact, most people believe that the biggest threat to humanity is some out of this world, unimaginable thing that will wipe out humans in one sweep. Well… the greatest threat to humanity is right under our noses." />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://vanshs.com/blog/technical/tcellsequencingneuralnetworks/" />
<meta property="article:published_time" content="2020-04-20T00:00:00+00:00" />
<meta property="article:modified_time" content="2020-04-20T00:00:00+00:00" /><meta property="og:site_name" content="Vansh" />






  </head>
  <body class="">
    <div class="container">
      <header class="header">
  <span class="header__inner">
    <a
  href="/"
  class="logo"
  style="text-decoration: none;"
>
  
    <span class="logo__mark"><svg xmlns="http://www.w3.org/2000/svg" class="greater-icon" viewBox="0 0 44 44">
  <path fill="none" d="M15 8l14.729 14.382L15 35.367" />
</svg>
</span>
    <span class="logo__text"
      >Vansh Sethi&#39;s Blog</span
    >
    <span class="logo__cursor"></span>
  
</a>

    <span class="header__right">
      
        <nav class="menu">
  <ul class="menu__inner menu__inner--desktop">
    
      
        
          <li><a href="/blog/technical">Technical Articles</a></li>
        
      
        
          <li><a href="/blog/thoughts">Thoughts</a></li>
        
      
      
      
  </ul>

  <ul class="menu__inner menu__inner--mobile">
    
      
        <li><a href="/blog/technical">Technical Articles</a></li>
      
    
      
        <li><a href="/blog/thoughts">Thoughts</a></li>
      
    
  </ul>
</nav>

        <span class="menu-trigger">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M0 0h24v24H0z" fill="none" />
            <path d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
          </svg>
        </span>
      
      <span class="theme-toggle">
        <svg
  class="theme-toggler"
  width="24"
  height="24"
  viewBox="0 0 48 48"
  fill="none"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M22 41C32.4934 41 41 32.4934 41 22C41 11.5066 32.4934 3 22
  3C11.5066 3 3 11.5066 3 22C3 32.4934 11.5066 41 22 41ZM7 22C7
  13.7157 13.7157 7 22 7V37C13.7157 37 7 30.2843 7 22Z"
  />
</svg>

      </span>
    </span>
  </span>
</header>


      <div class="content">
        
  
  

  <div class="post">
    <h1 class="post-title">T-Cell Receptor Sequencing using Neural Networks</h1>
    <div class="post-meta">
      
        <span class="post-date">
          2020-04-20
        </span>

        
      

      


      
    </div>

    

    

    <div class="post-content">
      
      <hr>
<h2 id="the-greatest-threat-to-humanity"><strong>The Greatest Threat to Humanity…</strong></h2>
<p>What is the greatest threat to humanity? Some might say that it’s a nuclear fallout. Others might say it’s the threat of a large asteroid hitting our planet. In fact, most people believe that the biggest threat to humanity is some out of this world, unimaginable thing that will wipe out humans in one sweep. Well… the greatest threat to humanity is right under our noses.</p>
<p>The greatest threat to humanity is a pandemic that could ravage around the world that we aren’t prepared for. Don’t believe me? Just look at what has happened in the past historically. The Spanish Flu inflected a quarter of the world’s population and killed upwards of 100 million people. Smallpox killed around 500 million people. Disease is the leading cause of death. Even billionaire philanthropist Bill Gates believes that the next worst thing to happen to humanity will be a virus.</p>
<blockquote>
<p><em>“If anything kills over 10 million people in the next few decades, it’s most likely to be a highly infectious virus rather than a war — not missiles, but microbes”</em></p>
</blockquote>
<p>Most of these viruses attack our immune system and terrorize our bodies from the inside. Fundamentally, we have been building vaccines as the main method of our attack to fight against these viruses. The way these vaccines work is that we inject the proteins of the virus into a person’s immune system to expose the immune system on how to fight against these viruses.</p>
<p>The fact is creating a vaccine is costly and takes a lot of time to create. Instead of having to expose the virus in a controlled matter to our immune system, what if we could probe our immune system to fight off these diseases without the use of a vaccine?</p>
<hr>
<h2 id="welcome-to-t-cells"><strong>Welcome to T-Cells</strong></h2>
<p>T-cells are the immune system’s natural defense against disease. They are cells within the immune system that identify dangerous cells (like cancer or viruses) and deactivate those cells via apoptosis. Here’s how they work:</p>
<p>Every diseased cell has antigens on its surface which are cultivated from the cell hijacking other cells and doing damage to other cells. T-Cells have receptors (TCRs) on the surface of their cell which can bind to certain antigens, which allow the T-cell to kill off the diseased cell. Your body does this all the time with diseases, however, sometimes foreign diseases adapt to your body’s natural TCRs which make it so they can’t bind with the diseased antigens. This becomes problematic as the diseased cell can spread throughout your body and essentially kill you. CAR T-cell therapies solve this issue by engineering specific TCRs to fight these antigens.</p>
<h3 id="car-t-cell-therapies"><strong>Car T-Cell Therapies</strong></h3>
<p>In a CAR T-cell therapy, a patient’s T-cells are extracted from a blood sample and genetic information is injected into the nucleus of the T-Cell to instruct the cell to produce certain T-cell receptors. These new CAR T-cells are then put back into the body to bind with the antigens to kill diseased cells.</p>
<p><img src="https://cdn-images-1.medium.com/max/2000/1*ogARkcNALfyCXIpvevniww.png" alt=""></p>
<p>When considering the TCR of the new T-Cell, there are a few major parts of the receptor to engineer to make sure that the antigen can bind with the TCR. The major parts include:</p>
<p><strong>The Antigen:</strong> Also referred to as the peptide, this protein sequence is what connects the TCR to the MHC (or connector to cell) to allow for apoptosis to occur.</p>
<p><strong>V-Segment:</strong> The V-segment or variable-segment is a part of the T-cell receptor and is the most physically outside part. There are many different variants of the v-segment.</p>
<p><strong>J-Segment:</strong> The J-segment or joining-segment is what connects the v-segment to the rest of the TCR. There are many different variants of the j-segment.</p>
<p><strong>CDR3:</strong> The 3rd complementarity-determining region of the T-cell is also important when fighting diseases. It’s in the form of a protein sequence.</p>
<p><strong>Epitope:</strong> The epitope is what joins the antigen to the antibody or the rest of the cell. It’s represented in the form of a sequence of proteins.</p>
<p>Finding the parts of the T-Cell receptor is called <strong>T-Cell Receptor Sequencing</strong> or TCR sequencing. Currently, finding the various parts of the TCR require lots of research and tests. It can be costly and timely, not ideal when dealing with a new disease. This is where CAR T-cell therapies become inefficient for new diseases. T-cell receptor sequencing is a hard process that equates many variables and is hard to do currently. <strong>I introduce a deep learning method to TCR sequencing, using a variety of deep learning methods.</strong></p>
<hr>
<h2 id="t-cell-receptor-sequencing-with-machine-learning"><strong>T-Cell Receptor Sequencing with Machine Learning</strong></h2>
<p><strong>Given the epitope protein sequence of an antigen, can we predict the V-Segment, J-Segment and CDR3 Protein sequence of the TCR that will bind with the antigen?</strong> This is the fundamental question I tried to answer using machine learning. To make this easier, I broke up the problem into three manageable goals. The first goal is to create a model to predict the V-segment, one model to predict the J-segment and one model to predict the CDR3 sequence.</p>
<p>I used the VDJdb database which contains information on epitopes and their corresponding T-cell parts that would trigger apoptosis. It contains over 75,000+ data points which is what the model was trained on. You can find the database <a href="https://vdjdb.cdr3.net/search">here</a>.</p>
<h2 id="deep-learning-model-for-v-segment-and-j-segment"><strong>Deep Learning Model for V-Segment and J-Segment</strong></h2>
<p>The method for predicting the v-segment and j-segment are very similar because they are only a handful of classes that the segments can be and a model can be tuned to predict the class that the segment is. <strong>This is a multiclass classification problem.</strong></p>
<h3 id="input-data-representation"><em>Input Data Representation</em></h3>
<p>As mentioned, the input of the neural networks are the epitope protein sequence of the antigen. However, the epitope protein sequence is represented as letters in a non-fixed length. This is problematic as neural networks only work exclusively with numerical values in a constant size input matter.</p>
<blockquote>
<p>Examples of epitope protein sequences: LLWNGPMAV (Yellow Fever Virus), CPSQEPMSIYVY (Cytomegalovirus), CTPYDINQM (Simian Immunodeficiency Virus)</p>
</blockquote>
<p>Luckily, assigning each letter to an id makes this easier. We can map each letter in the protein sequence to a number. For example, the letter “A” becomes 1, the letter “B” becomes 2 and so on.</p>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*CAbbXJt59qlWWyFc" alt=""></p>
<p>So the yellow fever virus sequence becomes:</p>
<blockquote>
<p>LLWNGPMAV → 12 12 23 14 7 16 13 1 22</p>
</blockquote>
<p>This new encoded sequence of numbers has a length of 9. However, some sequences will have a length of 8, 10, 11 or up to 20 proteins. The input for a neural network needs a fixed-sized input, so to achieve this, we can pad every sequence to the max length possible of 20 with 0’s. So for our currently encoded protein sequence of the yellow fever virus, it becomes:</p>
<blockquote>
<p>12 12 23 14 7 16 13 1 22 → 12 12 23 14 7 16 13 1 22 0 0 0 0 0 0 0 0 0 0 0</p>
</blockquote>
<p>These inputs are fed into our neural network in the form of arrays. So with the input handled, how does the neural network decide which segment is appropriate for the sequence?</p>
<h3 id="model-architecture"><em>Model Architecture</em></h3>
<p>A neural network is just a mathematical function that takes in input “x” and produces an output “y”. It has weights or learnt parameters that alter the x to get the y. There are different methodologies in neural networks that utilize parameters in different ways. These parameters are optimized to produce the best output possible. This is done by calculating a loss function which represents how good the model is. The lower the loss, the better the model. We can use an optimizer (which are calculus functions) to optimize the loss and make it lower. This process is called “machine learning” &amp; “training the model.”</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-python" data-lang="python">Input_Layer <span style="color:#f92672">=</span> tf<span style="color:#f92672">.</span>keras<span style="color:#f92672">.</span>layers<span style="color:#f92672">.</span>Input(shape<span style="color:#f92672">=</span>(<span style="color:#ae81ff">20</span>))

Embedding_Layer <span style="color:#f92672">=</span> tf<span style="color:#f92672">.</span>keras<span style="color:#f92672">.</span>layers<span style="color:#f92672">.</span>Embedding(<span style="color:#ae81ff">20</span>,<span style="color:#ae81ff">128</span>)(Input_Layer)
LSTM1 <span style="color:#f92672">=</span> tf<span style="color:#f92672">.</span>keras<span style="color:#f92672">.</span>layers<span style="color:#f92672">.</span>LSTM(<span style="color:#ae81ff">128</span>)(Embedding_Layer)

Dense0 <span style="color:#f92672">=</span> tf<span style="color:#f92672">.</span>keras<span style="color:#f92672">.</span>layers<span style="color:#f92672">.</span>Dense(<span style="color:#ae81ff">5000</span>,activation<span style="color:#f92672">=</span><span style="color:#e6db74">&#39;relu&#39;</span>)(LSTM1)
Dense1 <span style="color:#f92672">=</span> tf<span style="color:#f92672">.</span>keras<span style="color:#f92672">.</span>layers<span style="color:#f92672">.</span>Dense(<span style="color:#ae81ff">5000</span>,activation<span style="color:#f92672">=</span><span style="color:#e6db74">&#39;relu&#39;</span>)(Dense0)
Dense2 <span style="color:#f92672">=</span> tf<span style="color:#f92672">.</span>keras<span style="color:#f92672">.</span>layers<span style="color:#f92672">.</span>Dense(<span style="color:#ae81ff">5000</span>,activation<span style="color:#f92672">=</span><span style="color:#e6db74">&#39;relu&#39;</span>)(Dense1)
Dense3 <span style="color:#f92672">=</span> tf<span style="color:#f92672">.</span>keras<span style="color:#f92672">.</span>layers<span style="color:#f92672">.</span>Dense(<span style="color:#ae81ff">2500</span>,activation<span style="color:#f92672">=</span><span style="color:#e6db74">&#39;relu&#39;</span>)(Dense2)
Dense4 <span style="color:#f92672">=</span> tf<span style="color:#f92672">.</span>keras<span style="color:#f92672">.</span>layers<span style="color:#f92672">.</span>Dense(<span style="color:#ae81ff">500</span>,activation<span style="color:#f92672">=</span><span style="color:#e6db74">&#39;relu&#39;</span>)(Dense3) 

Dense5 <span style="color:#f92672">=</span> tf<span style="color:#f92672">.</span>keras<span style="color:#f92672">.</span>layers<span style="color:#f92672">.</span>Dense(<span style="color:#ae81ff">126</span>,activation<span style="color:#f92672">=</span><span style="color:#e6db74">&#39;softmax&#39;</span>)(Dense4)
model <span style="color:#f92672">=</span> tf<span style="color:#f92672">.</span>keras<span style="color:#f92672">.</span>Model(inputs<span style="color:#f92672">=</span>Input_Layer,outputs<span style="color:#f92672">=</span>Dense5)
</code></pre></div><p>The model has over 30 million parameters that can be optimized. It has standard dense layers that perform the most basic neural network operations on an input yet is very effective.</p>
<h3 id="output-data"><em>Output Data</em></h3>
<p>The last dense layer in the neural network has 126 neutrons which represent the 126 classes the V-segment of the T-Cell can be. The output is in the form of a one-hot vector which means every neuron output is 0 except for one of the neurons which is 1. The one neuron’s position that has a value of 1 determines which class the V-segment is.</p>
<p><img src="https://cdn-images-1.medium.com/max/2028/0*u9PHqpl5zp3KP2le" alt=""></p>
<p>For the J-segment model, there were 68 neurons that represent the 68 classes the J-segment can be. It’s the same model with just a different number of classes in the last dense layer.</p>
<p>So for example, if we were predicting the V-segment class, if the output of the model is 1 0 0 0…. it means that the V-segment would be the 1st class or the TRBV6–8 variable segment.</p>
<h3 id="training--results"><em>Training / Results</em></h3>
<p>After training both models, similar results were achieved. The loss went down to 3 for the entire dataset yet the validation loss was much worse. The model had a tough time generalizing on new data yet it works in theory. This is because many classes will work for a certain epitope so the loss in practicality is much lower.</p>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*7PFvseH_1_C3mxTR" alt="Loss for V-segment"></p>
<p><em>Loss for V-segment</em></p>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*iynn486k5ls-vzZT" alt="Loss for J-segment"></p>
<p><em>Loss for J-segment</em></p>
<p>After training for 30 epochs each, the model works decently well on new unseen data. I can predict one of the right classes up to 80% of the time.</p>
<hr>
<h2 id="sequence-2-sequence-for-cdr3-protein-sequence"><strong>Sequence 2 Sequence for CDR3 Protein Sequence</strong></h2>
<p>Since the CDR3 part of the T-cell is a protein sequence of a non-fixed length, a multiclass approach wouldn’t work. Instead, a new approach that takes in the epitope and produces an output of a dynamic length is needed. A Sequence 2 Sequence model can be used for this. It takes in a sequence (epitope protein sequence) and produces another sequence (cdr3 protein sequence). It incorporates an encoder-decoder model.</p>
<h3 id="input-data-representation-1"><em>Input Data Representation</em></h3>
<p>The way we represent the data for this model’s input is similar to the V-segment and J-segment models. The proteins in a sequence are mapped to a number. “A” is mapped to 1, “B” is mapped to 2 and so on. Yet, where the model’s input differs is that after encoding the proteins to numbers, those numbers are encoded into a one-hot vector. A one-hot vector is an array of 0’s with one number in the array as a 1. The position of the 1 in the array represents the class of that one-hot vector. For example, if the number 1 is encoded into a one-hot vector, the first position of the array would be 1 and the rest of the numbers would be 0. Since there are characters, there are 26 classes in the one-hot vector. Doing this allows for information to be handled easier through this neural network.</p>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*oBdgn1e4ITC7wHPx" alt=""></p>
<p>Also, at the beginning of the protein sequence and end, there will be a start and end token. These tokens essentially represent what they sound like, they indicate when a sequence is starting or when a sequence is ended. Also, the desired output will also be used during training explicitly within the model. I know, using the output for input? Yes, because it allows the seq2seq model to become more accurate and better.</p>
<h3 id="model-architecture-1"><em>Model Architecture</em></h3>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-python" data-lang="python">num_encoder_tokens <span style="color:#f92672">=</span> <span style="color:#ae81ff">41</span>
num_decoder_tokens <span style="color:#f92672">=</span> <span style="color:#ae81ff">41</span>
latent_dimension <span style="color:#f92672">=</span> <span style="color:#ae81ff">40</span>
<span style="color:#75715e"># Define an input sequence and process it.</span>
encoder_inputs <span style="color:#f92672">=</span> Input(shape<span style="color:#f92672">=</span>(None, num_encoder_tokens))
encoder <span style="color:#f92672">=</span> LSTM(latent_dimension, return_state<span style="color:#f92672">=</span>True)
encoder_outputs, state_h, state_c <span style="color:#f92672">=</span> encoder(encoder_inputs)

encoder_states <span style="color:#f92672">=</span> [state_h, state_c]

decoder_inputs <span style="color:#f92672">=</span> Input(shape<span style="color:#f92672">=</span>(None, num_decoder_tokens))

decoder_lstm <span style="color:#f92672">=</span> LSTM(latent_dimension, return_sequences<span style="color:#f92672">=</span>True, return_state<span style="color:#f92672">=</span>True)
decoder_outputs, _, _ <span style="color:#f92672">=</span> decoder_lstm(decoder_inputs, initial_state<span style="color:#f92672">=</span>encoder_states)
decoder_dense <span style="color:#f92672">=</span> Dense(num_decoder_tokens, activation<span style="color:#f92672">=</span><span style="color:#e6db74">&#39;softmax&#39;</span>)
decoder_outputs <span style="color:#f92672">=</span> decoder_dense(decoder_outputs)

model <span style="color:#f92672">=</span> Model([encoder_inputs, decoder_inputs], decoder_outputs)
</code></pre></div><p><strong>Embeddings:</strong> First, the input is put through an embedding layer, which essentially converts it into a new vector of values of a fixed length. This is important as we had to pad our input to keep it to a constant length and this process allows us to make the padded areas unimportant. It multiplies the input by some learned weights to create a new vector which are trained during the training process. Embeddings also allow for similar words or sequences to be generalized into similar number sequences, which allows the model to generalize even easier. Similar words are closer to each other.</p>
<p><img src="https://cdn-images-1.medium.com/max/3200/0*bCZRd3gGUmXvzCXQ" alt="Embeddings Map"><em>Embeddings Map</em></p>
<p><strong>Encoder:</strong> The encoder is an LSTM layer that produces the state vectors for the decoder. Essentially what the encoder is doing is that it processes the input and extracts the most important features. LSTMs have multiple state vectors that it aggregates, but essentially what the LSTM keeps track of is the long-term, short-term and produces an output. Propagating our input through this allows for crucial extraction of important features of our input, later to be constructed back into a sequence from the decoder.</p>
<p><strong>Decoder:</strong> The decoder is also an LSTM but reconstructs in some sense the output from the decoder. It is able to produce a representation of the new sequence from the encoder features. The decoder outputs are then run through a dense layer to produce the final sequence, which is encoded into numbers. These extremely specific neural network functionalities are what allows for the sequence 2 sequence model to be so effective. The parameters can be easily learnt with enough data.</p>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*4dLrNaHsNz2ZbMJu" alt=""></p>
<h3 id="training--results-1"><em>Training / Results</em></h3>
<p>This model trained very well and achieved magnificent results. There were only around 20 000 parameters (small for a neural network) but it was able to create a protein sequence to 90% accuracy and got down to a 0.4 loss.</p>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*1e7T_cWNIFKxiSr5" alt=""></p>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*kK-00WrbTIvJqNSm" alt=""></p>
<p>No signs of over fitting for the 20 epochs it was run for, which means it’s still got room for improvement! Overall, this works really well.</p>
<h2 id="demo"><strong>Demo</strong></h2>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*baDLDoOvnrDr-gF9.png" alt="Melanoma Antigen"></p>
<p><em>Melanoma Antigen</em></p>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*_dHUwO80LehOvNLu.png" alt="Herpes Virus Antigen"></p>
<p><em>Herpes Virus Antigen</em></p>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*7v9uGipjWW5Fg-DK.png" alt="Yellow Fever Virus Antigen"></p>
<p><em>Yellow Fever Virus Antigen</em></p>
<hr>
<h2 id="implications"><strong>Implications</strong></h2>
<p>I envision a future where pandemics, disease and viruses can be dealt with in a secure, hasty manner. We are inevitably going into a more personalized focused medical system that favours those with money over the general public. We as a society need to address large scale disease and make it accessible to everyone, and I believe that CAR T-cell therapy is something we need to be working to advance. Breakthroughs in research can expedite the process of getting it out widely to the market, but we will need full cooperation of world governments to achieve accessibility for everyone. This solution I’ve proposed helps in making CAR T-cell therapy better but not more accessible. As higher accuracy goes up, we need to ensure usability goes up as well.</p>
<p><img src="https://cdn-images-1.medium.com/max/2000/0*Vi82Ylq7_FCy9csO" alt=""></p>
<p>This project also helps cement the use of NLP technologies in the medical field (which is becoming more reliable). A lot of people don’t trust these A.I to help with medical discovery, but the fact is that it makes our current medical technology better.</p>
<p>Code for project: <a href="https://github.com/Vansh-Sethi/Predicting-Properties-of-T-Cells-to-Increase-Immune-Response">https://github.com/Vansh-Sethi/Predicting-Properties-of-T-Cells-to-Increase-Immune-Response</a></p>
<hr>
<h2 id="key-takeaways"><strong>Key Takeaways:</strong></h2>
<ul>
<li>
<p>Problem: Predict the CDR3 protein sequence, variable segment &amp; joining segment of a T-cell given the epitope of an antigen</p>
</li>
<li>
<p>Epitope data can be represent as number through word encodings and embeddings</p>
</li>
<li>
<p>Variable segment &amp; joining segment can be predicted using classical neural networks</p>
</li>
<li>
<p>CDR3 protein sequence can be predicted using a Sequence 2 Sequence model</p>
</li>
</ul>

    </div>
    

    
      
    
  </div>

      </div>

      
        <footer class="footer">
  <div class="footer__inner">
    
      <a
  href="/"
  class="logo"
  style="text-decoration: none;"
>
  
    <span class="logo__mark"><svg xmlns="http://www.w3.org/2000/svg" class="greater-icon" viewBox="0 0 44 44">
  <path fill="none" d="M15 8l14.729 14.382L15 35.367" />
</svg>
</span>
    <span class="logo__text"
      >Vansh Sethi&#39;s Blog</span
    >
    <span class="logo__cursor"></span>
  
</a>

      <div class="copyright">
        <span
          >© 2020 Powered by
          <a href="https://vanshs.com/" target="_blank" rel="noopener">My Thoughts</a></span
        >
        
      </div>
    
  </div>
</footer>

<script src="https://vanshs.com/blog/assets/main.js"></script>
<script src="https://vanshs.com/blog/assets/prism.js"></script>


      
    </div>

    
  </body>
</html>
